### DISCUSSION TITLE
WebSocket: Auto ping/pong?

### DISCUSSION THREAD

**Discussion by FGasper** [mojolicious/mojo#1998] [Q&A] [answered]
Hello,

Does Mojo’s WebSocket implementation include automatic ping/pong?

i.e., if nothing is sent nor received for, say, 5 minutes, does Mojo automatically send a ping frame and expect a pong in response? Or is that something Mojo callers need to implement manually?

Thank you!

**Comment by rabbiveesh** [comment, marked as answer]
The server will reply pong to a ping, but it doesn't do keep alive by
default.

Every case is different, sometimes you want keepalive on the client, and
sometimes on the server.

**Comment by FGasper** [comment]
>Every case is different, sometimes you want keepalive on the client, and sometimes on the server.

Thank you for clarifying. I guess if I send the ping manually, then I also have to check for the pong, ensure that the response matches, etc.?

**Comment by FGasper** [comment]
(Also, this is on the client.)
